crates/forge/bin/cmd/inspect.rs

```diff
@@ -69,6 +74,73 @@ impl InspectArgs {
         } else {
             project.find_contract_path(&contract.name)?
         };
+
+        // NOTE(zk) Fields that should have specific behavior for zksolc
+        // TODO: we should migrate all fields from fields_zksolc_unimplemented_warn to this array
+        // and eventually
+        let fields_zksolc_specific_behavior = vec![
+            ContractArtifactField::Abi,
+            ContractArtifactField::Bytecode,
+            ContractArtifactField::DeployedBytecode,
+        ];
+
+        let fields_zksolc_unimplemented_warn = vec![
+            ContractArtifactField::GasEstimates,
+            ContractArtifactField::StorageLayout,
+            ContractArtifactField::Metadata,
+            ContractArtifactField::Eof,
+            ContractArtifactField::EofInit,
+        ];
+
+        // We should not try to implement zk behavior for this
+        let fields_zksolc_should_error = vec![
+            ContractArtifactField::Assembly,
+            ContractArtifactField::AssemblyOptimized,
+            ContractArtifactField::LegacyAssembly,
+            ContractArtifactField::Ir,
+            ContractArtifactField::IrOptimized,
+            ContractArtifactField::Ewasm,
+        ];
+
+        // NOTE(zk): ZKsync version of inspect does not support all fields
+        if modified_build_args.compiler.zk.enabled() && fields_zksolc_should_error.contains(&field)
+        {
+            return Err(eyre::eyre!("ZKsync version of inspect does not support this field"))
+        } else if modified_build_args.compiler.zk.enabled() &&
+            fields_zksolc_unimplemented_warn.contains(&field)
```

Comment:
```suggestion
        }
        if modified_build_args.compiler.zk.enabled() &&
            fields_zksolc_unimplemented_warn.contains(&field)
```
Don't need `else` if returning